node-backend/app/controllers/v1/recognitionController.js

```diff
@@ -0,0 +1,346 @@
+        });
+      } else if (data.dataValues.org_id == tokenData.orgId) {
+        return true;
+      } else {
+        res.status(404).send({
+          error: {
+            message: "User not found in specified organisation",
+          },
+        });
+      }
+    })
+    .catch((err /*eslint-disable-line no-unused-vars*/) => {
+      res.status(500).send({
+        error: {
+          message: "internal server error",
+        },
+      });
+    });
+};
+
+const addRecognition = (req, res, recognitions) => {
+  Recognitions.create(recognitions)
+    .then((info) => {
+      res.status(201).send({
+        data: info,
+      });
+    })
+    .catch((err /*eslint-disable-line no-unused-vars*/) => {
+      res.status(500).send({
+        error: {
+          message: "internal server error",
+        },
+      });
+    });
+};
+
+module.exports.create = async (req, res) => {
+  const tokenData = await jwtValidate.getData(req.headers["authorization"]);
+  // Create a Recognition
+  const recognitions = {
+    core_value_id: req.body.core_value_id,
+    text: req.body.text,
+    given_for: req.body.given_for,
+    given_by: tokenData.userId,
+    given_at: moment.utc().unix(),
+  };
+  // Validate request
+  schema
+    .validate(recognitions, { abortEarly: false })
+    .then(async () => {
+      if (await validateCoreValue(req, res)) {
+        if (await validateGivenFor(req, res)) {
+          await addRecognition(req, res, recognitions);
+        }
+      }
+    })
+    .catch((err) => {
+      res.status(400).send({
+        error: utility.getFormattedErrorObj(
+          "invalid recognition",
+          "invalid recognition Data",
+          err.errors
+        ),
+      });
+    });
+};
+
+module.exports.findOne = async (req, res) => {
+  const idSchema = yup.object().shape({
+    id: yup
+      .number()
+      .required({ id: "required" })
+      .typeError({ id: "should be number " }),
+  });
+
+  idSchema
+    .validate({ id: req.params.id }, { abortEarly: false })
+    .then(() => {
+      Recognitions.findByPk(req.params.id)
+        .then((data) => {
+          if (data == null /*eslint-disable-line no-eq-null*/) {
+            res.status(404).send({
+              error: {
+                message: "Recognition with specified id is not found",
+              },
+            });
+          } else {
+            res.status(200).send({
+              data: data,
+            });
+          }
+        })
+        .catch(() => {
+          res.status(500).send({
+            error: {
+              message: "internal server error ",
+            },
+          });
+        });
+    })
+    .catch((err) => {
+      res.status(400).send({
+        error: utility.getFormattedErrorObj(
+          "invalid recognition",
+          "invalid recognition Data",
+          err.errors
+        ),
+      });
+    });
+};
+
+const createFilterQuery = (filterData, tokenData) => {
+  const sqlQuery = "select * from recognitions where given_for";
```

Comment:
select * from recognitions should be your base string. And as per the query params, you should construct where clause and at the end you can concat it to base query

Reply:
yes